diff --git a/src/genl/genl_qer.c b/src/genl/genl_qer.c
index 68c37db..b0d377d 100644
--- a/src/genl/genl_qer.c
+++ b/src/genl/genl_qer.c
@@ -315,13 +315,15 @@ static int qer_fill(struct qer *qer, struct gtp5g_dev *gtp, struct genl_info *in
             qer->mbr.ul_high = nla_get_u32(mbr_param_attrs[GTP5G_QER_MBR_UL_HIGH32]);
             qer->mbr.ul_low  = nla_get_u8(mbr_param_attrs[GTP5G_QER_MBR_UL_LOW8]);
             qer->ul_mbr = concat_bit_rate(qer->mbr.ul_high, qer->mbr.ul_low);
-            qer->ul_policer = newTrafficPolicer(qer->ul_mbr);
+            if (qer->ul_mbr > 0)
+                qer->ul_policer = newTrafficPolicer(qer->ul_mbr);
         }
         if (mbr_param_attrs[GTP5G_QER_MBR_DL_HIGH32] &&
             mbr_param_attrs[GTP5G_QER_MBR_DL_LOW8]) {
             qer->mbr.dl_high = nla_get_u32(mbr_param_attrs[GTP5G_QER_MBR_DL_HIGH32]);
             qer->mbr.dl_low  = nla_get_u8(mbr_param_attrs[GTP5G_QER_MBR_DL_LOW8]);
             qer->dl_mbr = concat_bit_rate(qer->mbr.dl_high, qer->mbr.dl_low);
-            qer->dl_policer = newTrafficPolicer(qer->dl_mbr);
+            if (qer->dl_mbr > 0)
+                qer->dl_policer = newTrafficPolicer(qer->dl_mbr);
         }
     }
